aea/aea_builder.py

```diff
@@ -0,0 +1,418 @@
+        return len(not_supported_packages) == 0
+
+    @contextmanager
+    def load_dependencies(self):
+        """
+        Load dependencies of a component, so its modules can be loaded.
+
+        :return: None
+        """
+        modules = self._get_import_order()
+        with _SysModules.load_modules(modules):
+            yield
+
+    def _get_import_order(self) -> List[Tuple[str, types.ModuleType]]:
+        """
+        Get the import order.
+
+        At the moment:
+        - a protocol has no dependencies.
+        - a connection can depend on protocols.
+        - a skill can depend on protocols.
+        - a contract ...
+
+        :return: a list of pairs: (import path, module object)
+        """
+        # get protocols first
+        protocols = [
+            (self.build_dotted_part(component, relative_import_path), module_obj)
+            for component in self.protocols.values()
+            for (
+                relative_import_path,
+                module_obj,
+            ) in component.importpath_to_module.items()
+        ]
+        connections = [
+            (self.build_dotted_part(component, relative_import_path), module_obj)
+            for component in self.connections.values()
+            for (
+                relative_import_path,
+                module_obj,
+            ) in component.importpath_to_module.items()
+        ]
+        skills = [
+            (self.build_dotted_part(component, relative_import_path), module_obj)
+            for component in self.skills.values()
+            for (
+                relative_import_path,
+                module_obj,
+            ) in component.importpath_to_module.items()
+        ]
+        return cast(
+            List[Tuple[str, types.ModuleType]], protocols + connections + skills
+        )
+
+    def build_dotted_part(self, component, relative_import_path) -> str:
+        """Given a component, build a dotted path for import."""
+        if relative_import_path == "":
+            return component.prefix_import_path
+        else:
+            return component.prefix_import_path + "." + relative_import_path
+
+
+class AEABuilder:
+    """This class helps to build an AEA."""
+
+    def __init__(self):
+        self._resources = Resources()
+
+        # identity
+        self._name = None
+        self._addresses = {}  # type: Dict[str, Address]
+        self._default_key = None  # set by the user, or instantiate a default one.
+
+        self._package_dependency_graph = _DependenciesManager()
+
+        # add default protocol
+        self.add_protocol(Path(AEA_DIR, "protocols", "default"))
+        # add stub connection
+        self.add_connection(Path(AEA_DIR, "connections", "stub"))
+        # add error skill
+        self.add_skill(Path(AEA_DIR, "skills", "error"))
+
+    def set_name(self, name: str) -> "AEABuilder":
+        self._name = name
+        return self
+
+    def add_address(self, identifier: str, address: Address) -> "AEABuilder":
```

Comment:
I think the `add_address` should be replaced with an `add_private_key`. As all addresses are derived from a private key (via the wallet)

Reply:
And it should consume a file_path to the private key file

Reply:
Then, when I add encryption of the private key file, we can add a password without further changes to the builder